lnwire/channel_ready.go

```diff
@@ -38,6 +47,16 @@ type ChannelReady struct {
 	// ShortChannelID for forwarding.
 	AliasScid *ShortChannelID
 
+	// AnnouncementBitcoinNonce is an optional field that stores a public
+	// nonce that will be used along with the node's bitcoin key during
+	// signing of the ChannelAnnouncement2 message.
+	AnnouncementBitcoinNonce fn.Option[Musig2Nonce]
```

Comment:
Can be `fn.OptionalRecordT`. Then no need for the variables above for each type. Also this way, looking at the definition of the attribute, you already know what the TLV type will be. 

Once we merge https://github.com/lightningnetwork/lnd/pull/8334 then we can start to use it here.